docs/developer/serving/traffic-management.md

```diff
@@ -0,0 +1,220 @@
+
+## Traffic routing examples
+
+The following example shows a `traffic` spec where 100% of traffic is routed to the `latestRevision` of the Service. Under `status` you can see the name of the latest Revision that `latestRevision` was resolved to:
+
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: example-service
+  namespace: default
+spec:
+...
+  traffic:
+  - latestRevision: true
+    percent: 100
+status:
+  ...
+  traffic:
+  - percent: 100
+    revisionName: example-service-1
+```
+
+The following example shows a `traffic` spec where 100% of traffic is routed to the `current` Revision, and the name of that Revision is specified as `example-service-1`. The latest ready Revision is kept available, even though no traffic is being routed to it:
+
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: example-service
+  namespace: default
+spec:
+...
+  traffic:
+  - tag: current
+    revisionName: example-service-1
+    percent: 100
+  - tag: latest
+    latestRevision: true
+    percent: 0
+```
+
+The following example shows how the list of Revisions in the `traffic` spec can be extended so that traffic is split between multiple Revisions. This example sends 50% of traffic to the `current` Revision, `example-service-1`, and 50% of traffic to the `candidate` Revision, `example-service-2`:
+
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: example-service
+  namespace: default
+spec:
+...
+  traffic:
+  - tag: current
+    revisionName: example-service-1
+    percent: 50
+  - tag: candidate
+    revisionName: example-service-2
+    percent: 50
+  - tag: latest
+    latestRevision: true
+    percent: 0
+```
+
+## Routing and managing traffic by using the Knative CLI
+
+You can use the following `kn` CLI command to split traffic between revisions:
+
+```bash
+kn service update <service-name> --traffic <revision-name>=<percent>
+```
+
+Where:
+
+- `<service-name>` is the name of the Knative Service that you are configuring traffic routing for.
+- `<revision-name>` is the name of the revision that you want to configure to receive a percentage of traffic.
+- `<percent>` is the percentage of traffic that you want to send to the revision specified by `<revision-name>`.
+
+For example, to split traffic for a Service named `example`, by sending 80% of traffic to the Revision `green` and 20% of traffic to the Revision `blue`, you could run the following command:
+
+```bash
+kn service update example --traffic green=80 --traffic blue=20
+```
+
+## Routing and managing traffic with blue/green deployment
+
+You can safely reroute traffic from a live version of an application to a new version by using a [blue/green deployment strategy](https://en.wikipedia.org/wiki/Blue-green_deployment).
+
+### Procedure
```

Comment:
@abrennan89 I'm fine with the documentation, that is copied over, but for the long term we should definitely describe it in `kn` terms. The example does not have a smooth UX (cryptic kubectl calls, editing yaml files) and kn should definitely cover this use case (and if not, we need to fix kn)

Reply:
I created https://github.com/knative/docs/issues/3951 to track this additional update.

Reply:
+1, for now I'm just moving it for now and removing unnecessary files etc, but it would be nice to clean this up properly when the UX is figured out.

Reply:
Yes, I understand this and it looks much better now !